text/0000-ember-language-tooling-roadmap.md

```diff
@@ -0,0 +1,169 @@
+
+I will leave this part underspecified for now -- it'll be easier to elaborate on what needs to happen here after we've completed our work to re-work Glint atop of Volar.
+
+### Proposal 6: Move Glint to the "ember-tooling" Github Organization.
+
+Regardless of how Proposal 5 is resolved, we should consolidate Ember tooling into the "ember-tooling" organization, rather than having things sprawled between "ember-tooling" and "typed-ember".
+
+### Proposal 7: Strongly encourage the deprecation and removal of sprawling editor extensions with overlapping functionality in the ecosystem
+
+With a unified Glint + ELS, we encourage maintainers of extensions providing Ember syntaxes / tooling to deprecate and ultimately remove their extensions from the VSCode marketplace and archive their Github repos, directing their traffic to the Ember Language Tooling package.
+
+## How we teach this
+
+(This RFC section not applicable.)
+
+## Drawbacks
+
+There is some risk to placing such a heavy bet on Volar in case they take things in a direction that is not advisable or incompatible/inconvenient to Ember. If Volar takes things in an unseemly direction, we would have to use and maintain a fork of Volar.
+
+## Alternatives
+
+The alternative is to continue to add features to Glint to achieve feature parity with vanilla TS, but ultimately this would mean duplicating the work of Volar and any other language tooling that isn't dependent on Volar, while probably doing a worse job implementing these features in the process.
+
+Also, "alternatively" we could still pursue merging ELS + Glint and punt on the Volar refactor, which would still fix some points of confusion and complexity in Ember's tooling ecosystem.
+
+## Unresolved questions
+
+### Support class-backed components?
+
+Continuing to support the classic two-file approach to Ember components (optional .js/.ts component file backing the .hbs template) will be more difficult than only supporting .gjs/.gts files (in fact one of the motivators for the .gts format was to avoid the complexity/ambiguity of the multi-file approach). I have not reached the point of the Volar spike to know how practical/possible it is to represent the classic two file approach.
+
+There have been some discussions in the community as to whether we should just cut ties with the two-file past, but given that language tooling for .gts/.gjs is not that great, there's been less motivation for people (including this RFC's author) to migrate their components to .gts until tooling "is ready", and even when it is ready, it's still a lot of work for companies to migrate their code to .gts.
```

Comment:
We discussed this and my vote is that any new tooling is only focused on the future and happy path. People still needing to support older apps can use the older tooling.

> it's still a lot of work for companies to migrate their code to .gts.

This burden should be eased once @void-mAlex finishes his codemod

Reply:
It will be easier to achieve the .gts/.gjs support milestone as its own unit of work, regardless of class-backed component support, so maybe we can revisit that later. Maybe I can update the RFC to separate gts support out as a more agreeable preliminary milestone.